### ISSUE TITLE
Ruby 2.7.1 undefined method `file=' with livereload

### ISSUE THREAD

**Issue by dml-1** [guard/guard#964]
`bundle exec guard` => 

```bash
15:15:25 - ERROR - Guard::LiveReload failed to achieve its <stop>, exception was:
> [#9731be21afc2] NoMethodError: undefined method `stop' for nil:NilClass
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-livereload-2.5.2/lib/guard/livereload.rb:36:in `stop'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:83:in `block (2 levels) in _supervise'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/ui/config.rb:62:in `block in with_progname'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/lumberjack-1.0.13/lib/lumberjack/logger.rb:271:in `push_thread_local_value'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/lumberjack-1.0.13/lib/lumberjack/logger.rb:232:in `set_progname'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/ui/config.rb:61:in `with_progname'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:81:in `block in _supervise'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:79:in `catch'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:79:in `_supervise'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:22:in `block (3 levels) in run'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:121:in `block (2 levels) in _run_group_plugins'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:119:in `each'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:119:in `block in _run_group_plugins'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:118:in `catch'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:118:in `_run_group_plugins'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:21:in `block (2 levels) in run'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:20:in `each'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:20:in `block in run'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/lumberjack-1.0.13/lib/lumberjack.rb:34:in `unit_of_work'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/runner.rb:18:in `run'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/commander.rb:59:in `stop'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/commander.rb:52:in `ensure in start'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/commander.rb:52:in `start'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/cli/environments/valid.rb:16:in `start_guard'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/cli.rb:122:in `start'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/aruba_adapter.rb:32:in `execute'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/aruba_adapter.rb:19:in `execute!'
> [#9731be21afc2] /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/bin/_guard-core:11:in `<main>'
15:15:25 - INFO - Guard::LiveReload has just been fired

15:15:25 - INFO - Bye bye...
/Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/jobs/pry_wrapper.rb:131:in `_setup': undefined method `file=' for #<Pry::History:0x00007ff432ad7220>
Did you mean?  filter (NoMethodError)
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/jobs/pry_wrapper.rb:68:in `initialize'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/interactor.rb:17:in `new'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/interactor.rb:17:in `initialize'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard.rb:67:in `new'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard.rb:67:in `setup'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/commander.rb:32:in `start'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/cli/environments/valid.rb:16:in `start_guard'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/cli.rb:122:in `start'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/aruba_adapter.rb:32:in `execute'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/lib/guard/aruba_adapter.rb:19:in `execute!'
	from /Users/dluan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/guard-2.16.1/bin/_guard-core:11:in `<main>'
```

Thank you for your tremendous work, just wanted to include this bug report in case others come across this.

**Comment by rymai** [comment]
@dml-1 Hi, I believe this has been solved in Guard 2.16.2 actually: https://github.com/guard/guard/releases/tag/v2.16.2 (this isn't a Ruby 2.7 error, but an incompatibility with Pry 0.13. 🙏

**Comment by rymai** [comment]
Same as https://github.com/guard/guard/issues/956, https://github.com/guard/guard/issues/953, and https://github.com/guard/guard/issues/957.
